File: Practica3/QytetetJava/src/modeloqytetet/Tablero.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.ArrayList;

/**
 *
 * @author Lorena
 */
public class Tablero {
   private ArrayList<Casilla> casillas;
   private Casilla carcel;
   
   //Constructores
   //Sin argumentos
   Tablero(){
       casillas = new ArrayList();
       inicializar();
   }
   
   //consultores
   Casilla getCarcel(){
       return carcel;
   }
   

    //Metodos 
    boolean esCasillaCarcel(int numeroCasilla){
        return numeroCasilla == getCarcel().getNumeroCasilla();
    }
    
    Casilla obtenerCasillaNumero(int numeroCasilla){
        Casilla obtCasilla = null;
        for(int i = 0; i < casillas.size(); i++){
            if(numeroCasilla == casillas.get(i).getNumeroCasilla()){
                obtCasilla =  casillas.get(i);                
            }
        }
        return obtCasilla;
    }
    
    Casilla obtenerNuevaCasilla(Casilla casilla, int desplazamiento){
        int nuevaCasilla = casilla.getNumeroCasilla() + desplazamiento;
        if (nuevaCasilla > 19){
            nuevaCasilla = (nuevaCasilla%19) - 1;
        }
        Casilla c = obtenerCasillaNumero(nuevaCasilla);
        return c;
    }
    
    //Método inicializar
    private void inicializar(){
        
        
        //Casilla de salida
        casillas.add(new Casilla(0, TipoCasilla.SALIDA));
        
        //Casillas de tipo calle
        casillas.add(new Casilla(1, 100, TipoCasilla.CALLE, new TituloPropiedad("Calle Piruleta", 60, (float) 0.1, 150, 250)));
        casillas.add(new Casilla(2, 300, TipoCasilla.CALLE, new TituloPropiedad("Calle Nube", 50, (float) 0.16, 156, 270)));
        casillas.add(new Casilla(3, 210, TipoCasilla.CALLE, new TituloPropiedad("Calle Gominola", 55, (float) 0.11, 160, 300)));
        
        //Casilla de tipo sopresa
        casillas.add(new Casilla(4, TipoCasilla.SORPRESA));
        
        //Casillas de tipo calle
        casillas.add(new Casilla(5, 400, TipoCasilla.CALLE, new TituloPropiedad("Calle de Chocolate", 70, (float) -0.1, 400, 300)));
        casillas.add(new Casilla(6, 450, TipoCasilla.CALLE, new TituloPropiedad("Calle Bombon", 75, (float) -0.16, 500, 350)));
        
        //Casilla tipo Juez
        casillas.add(new Casilla(7, TipoCasilla.JUEZ));
        
        //Casilla tipo calle
        casillas.add(new Casilla(8, 100, TipoCasilla.CALLE, new TituloPropiedad("Calle Perla", 50, (float) -0.2, 150, 250)));
        
        //Casilla tipo Carcel
        casillas.add(new Casilla(9, TipoCasilla.CARCEL));
        
        //Casilla tipo Sorpresa
        casillas.add(new Casilla(10, TipoCasilla.SORPRESA));
        
        //Casillas de tipo calle
        casillas.add(new Casilla(11, 470, TipoCasilla.CALLE, new TituloPropiedad("Calle Bastón de Caramelo", 70, (float) 0.1, 600, 300)));
        casillas.add(new Casilla(12, 480, TipoCasilla.CALLE, new TituloPropiedad("Calle Cupcake", 75, (float) 0.16, 700, 350)));
        casillas.add(new Casilla(13, 490, TipoCasilla.CALLE, new TituloPropiedad("Calle Manzana de Caramelo", 80, (float) 0.17, 750, 400)));
        
        //Casilla tipo Sorpresa
        casillas.add(new Casilla(14, TipoCasilla.SORPRESA));
        
        //Casilla tipo Parking
        casillas.add(new Casilla(15, TipoCasilla.PARKING));
        
        //Casillas de tipo calle
        casillas.add(new Casilla(16, 600, TipoCasilla.CALLE, new TituloPropiedad("Calle Galleta", 80, (float) 0.18, 800, 560)));
        casillas.add(new Casilla(17, 650, TipoCasilla.CALLE, new TituloPropiedad("Calle Bizcocho", 90, (float) 0.19, 900, 600)));
        
        //Casilla tipo Impuesto
        casillas.add(new Casilla(18, TipoCasilla.IMPUESTO));
        
        //Casillas de tipo calle
        casillas.add(new Casilla(19, 700, TipoCasilla.CALLE, new TituloPropiedad("Calle Lemon Pie", 100, (float) 0.2, 1000, 750)));
        
        //Inicializar casilla carcel en el numero 9
        carcel = casillas.get(9);
        
    }
    
    //Método toString() devuelve un String con el estado del objeto correspondiente
    @Override
    public String toString(){
        return "Casillas " + casillas.toString() 
               + "\n Carcel en" + carcel + "\n";
    }
}

File: Practica1/Examen/QytetetRubyPruebaExamen/lib/tablero.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'tipo_casilla'
require_relative 'casilla'

module ModeloQytetet
  class Tablero
    #Consultor
    attr_reader :carcel

    #Constructor
    def initialize()
      @casillas = Array.new
      inicializar
    end
    
    #Metodos
    def es_casilla_carcel (numeroCasilla)
      return false
    end
    
    def obtener_casilla_numero(numeroCasilla)
      return @casillas.at(numeroCasilla)
    end
    
    def obtener_nueva_casilla(casilla,desplazamiento)
      return nil
    end
    
    def inicializar
        
      
        #Casilla de salida
        @casillas << (Casilla.no_calle(0, TipoCasilla::SALIDA))
        
        #Casillas de tipo calle
        @casillas << (Casilla.new(1, 100, TipoCasilla::CALLE,  TituloPropiedad.new('Calle Piruleta', 60, 0.1, 150, 250)))
        @casillas << (Casilla.new(2, 300, TipoCasilla::CALLE,  TituloPropiedad.new('Calle Nube', 50,  0.16, 156, 270)))
        @casillas << (Casilla.new(3, 210, TipoCasilla::CALLE,  TituloPropiedad.new('Calle Gominola', 55,  0.11, 160, 300)))
        
        #Casilla de tipo sopresa
        @casillas << (Casilla.no_calle(4, TipoCasilla::SORPRESA))
        
        #Casillas de tipo calle
        @casillas << (Casilla.new(5, 400, TipoCasilla::CALLE,  TituloPropiedad.new('Calle de Chocolate', 70,  -0.1, 400, 300)))
        @casillas << (Casilla.new(6, 450, TipoCasilla::CALLE,  TituloPropiedad.new('Calle Bombon', 75,  -0.16, 500, 350)))
        
        #Casilla tipo Juez
        @casillas << (Casilla.no_calle(7, TipoCasilla::JUEZ))
        
        #Casilla tipo calle
        @casillas << (Casilla.new(8, 100, TipoCasilla::CALLE,  TituloPropiedad.new('Calle Perla', 50,  -0.2, 150, 250)))
        
        #Casilla tipo Carcel
        @casillas << (Casilla.no_calle(9, TipoCasilla::CARCEL))
        
        #Casilla tipo Sorpresa
        @casillas << (Casilla.no_calle(10, TipoCasilla::SORPRESA))
        
        #Casillas de tipo calle
        @casillas << (Casilla.new(11, 470, TipoCasilla::CALLE,  TituloPropiedad.new('Calle Baston de Caramelo', 70,  0.1, 600, 300)))
        @casillas << (Casilla.new(12, 480, TipoCasilla::CALLE,  TituloPropiedad.new('Calle Cupcake', 75,  0.16, 700, 350)))
        @casillas << (Casilla.new(13, 490, TipoCasilla::CALLE,  TituloPropiedad.new('Calle Manzana de Caramelo', 80, 0.17, 750, 400)))
        
        #Casilla tipo Sorpresa
        @casillas << (Casilla.no_calle(14, TipoCasilla::SORPRESA))
        
        #Casilla tipo Parking
        @casillas << (Casilla.no_calle(15, TipoCasilla::PARKING))
        
        #Casillas de tipo calle
        @casillas << (Casilla.new(16, 600, TipoCasilla::CALLE,  TituloPropiedad.new('Calle Galleta', 80, 0.18, 800, 560)))
        @casillas << (Casilla.new(17, 650, TipoCasilla::CALLE,  TituloPropiedad.new('Calle Bizcocho', 90, 0.19, 900, 600)))
        
        #Casilla tipo Impuesto
        @casillas << (Casilla.no_calle(18, TipoCasilla::IMPUESTO))
        
        #Casillas de tipo calle
        @casillas << (Casilla.new(19, 700, TipoCasilla::CALLE,  TituloPropiedad.new('Calle Lemon Pie', 100, 0.2, 1000, 750)))
        
        #Inicializar casilla carcel en el numero 9
        @carcel = @casillas.at(9)
    end
 
    #Método toString() devuelve un String con el estado del objeto correspondiente
    def to_s
      "Casillas #{@casillas.each {|i| puts i.to_s}}
       \nCarcel en: #{@carcel}"
    end
    
  end
end
